Fix(ui) miscenllaneous bugs

YvonTre opened the pull request:
- Fix TheProgressBar height problem.
- Fix getAwayImStarting button color flickering.
- Finish straightToTheYellowDragon functionality.
- Add nextVideo indicator icon.

Commit:
fix(mute): fix mute functionality won't work

Commit:
Merge branch 'develop' into fix(ui)-miscenllaneous-bugs

Commit:
Resolve all conflicts

Commit:
Merge branch 'fix(ui)-miscenllaneous-bugs' of https://github.com/YvonTre/splayerx into fix(ui)-miscenllaneous-bugs

Commit:
fix(ui): fix style-related TheProgressBar bugs

Commit:
fix(ui): fix leftFakeButton and play event trigger

Commit:
fix(ui): fix progress bar height changes and tests

Commit:
refactor(video): fix video cannot play and tweak finalPartTime

Commit:
Merge branch 'develop' into fix(ui)-miscenllaneous-bugs

Commit:
refactor(vuex): remove useless currentPlaying

ipy commented on src/renderer/components/PlayingView/TheProgressBar.vue:
```diff
@@ -94,13 +99,13 @@ export default {
     },
     leftFakeProgressBackgroundColor() {
-      let opacity = 0.908;
+      let opacity;
```
没有默认值的话，不是会出现 undefined 的情况？

YvonTre replied:
现在把默认值加回去了。

Commit:
fix(errorlog): fix all error logs

1. Vuex array and object cannot be mutate directly, immutability is must
2. Refactor AudioTrack switch logic.
3. Fix clear AudioTrackList bugs.

Commit:
feat(nextvideo): implement NextVideo function

Commit:
fix(ui): fix TheProgressBar's height problem

Commit:
feat(ui): add nextVideo indicator to ThePreviewThumbnail

Commit:
fix(ui): revert TheProgressBar height changes

Commit:
fix(ui): fix rightInvisible won't trigger thumbnail disappear and other

1. Fix when mouseleave from TheProgressBar, TheProgressBar won't go away

Commit:
test(fix): fix TheProgressBar test

Commit:
Resolve all conflicts.

Commit:
fix(ui): add default value to leftFakeProgressBackgroundColor

ipy commented on src/main/index.js:
```diff
@@ -223,9 +223,10 @@ function createWindow() {
 
   mainWindow.once('ready-to-show', () => {
     mainWindow.show();
+    const vidRegex = new RegExp('^(.3g2|.3gp|.3gp2|.3gpp|.amv|.asf|.avi|.bik|.bin|.crf|.divx|.drc|.dv|.dvr-ms|.evo|.f4v|.flv|.gvi|.gxf|.iso|.m1v|.m2v|.m2t|.m2ts|.m4v|.mkv|.mov|.mp2|.mp2v|.mp4|.mp4v|.mpe|.mpeg|.mpeg1|.mpeg2|.mpeg4|.mpg|.mpv2|.mts|.mtv|.mxf|.mxg|.nsv|.nuv|.ogg|.ogm|.ogv|.ogx|.ps|.rec|.rm|.rmvb|.rpl|.thp|.tod|.tp|.ts|.tts|.txd|.vob|.vro|.webm|.wm|.wmv|.wtv|.xesc)$');
```
正则的`.`可以匹配任意字符，需要转义

YvonTre replied:
请问哪里需要转义字符呢，在console里试过了并不是可以匹配任意字符的。
现在把`^`删掉，因为不是完全匹配后缀名。

ipy replied:
![default](https://user-images.githubusercontent.com/2088040/48761105-c9f76d80-ece2-11e8-99fb-37a3059bf69e.png)

YvonTre replied:
好的，现在改成这样了。
`const vidRegex = new RegExp('\\.(3g2|3gp|3gp2|3gpp|amv|asf|avi|bik|bin|crf|divx|drc|dv|dvr-ms|evo|f4v|flv|gvi|gxf|iso|m1v|m2v|m2t|m2ts|m4v|mkv|mov|mp2|mp2v|mp4|mp4v|mpe|mpeg|mpeg1|mpeg2|mpeg4|mpg|mpv2|mts|mtv|mxf|mxg|nsv|nuv|ogg|ogm|ogv|ogx|ps|rec|rm|rmvb|rpl|thp|tod|tp|ts|tts|txd|vob|vro|webm|wm|wmv|wtv|xesc)$');`

ipy commented on src/renderer/components/PlayingView/InfoCard.vue:
```diff
@@ -10,17 +10,17 @@ export default {
       type: Number,
       default: 1,
-      validator: value => value >= 1 && !Number.isFinite(value),
+      validator: value => value > 0 && !Number.isNaN(value),
```
什么情况下 `value > 0` is时 `Number.isNaN(value)` 为真呢？

YvonTre replied:
调整回`value > 0`了。

ipy commented on src/renderer/components/PlayingView/VideoCanvas.vue:
```diff
@@ -17,7 +17,7 @@
       :updateCurrentTime="true"
       :currentTime="seekTime"
-      :currentAudioTrackId="currentAudioTrackId"
+      :currentAudioTrackId="currentAudioTrackId.toString()"
```
这里有没有可能出现空引用？

YvonTre replied:
`currentAudioTrack`是vuex的getter，有默认返回值-1的，应该不会出现空引用。

ipy commented on src/renderer/main.js:
```diff
@@ -660,7 +627,7 @@ new Vue({
       const subRegex = new RegExp('^(.srt|.ass|.vtt)$');
       const videoFiles = [];
-      const vidRegex = new RegExp('^(3g2|.3gp|.3gp2|.3gpp|.amv|.asf|.avi|.bik|.bin|.crf|.divx|.drc|.dv|.dvr-ms|.evo|.f4v|.flv|.gvi|.gxf|.iso|.m1v|.m2v|.m2t|.m2ts|.m4v|.mkv|.mov|.mp2|.mp2v|.mp4|.mp4v|.mpe|.mpeg|.mpeg1|.mpeg2|.mpeg4|.mpg|.mpv2|.mts|.mtv|.mxf|.mxg|.nsv|.nuv|.ogg|.ogm|.ogv|.ogx|.ps|.rec|.rm|.rmvb|.rpl|.thp|.tod|.tp|.ts|.tts|.txd|.vob|.vro|.webm|.wm|.wmv|.wtv|.xesc)$');
+      const vidRegex = new RegExp('^(.3g2|.3gp|.3gp2|.3gpp|.amv|.asf|.avi|.bik|.bin|.crf|.divx|.drc|.dv|.dvr-ms|.evo|.f4v|.flv|.gvi|.gxf|.iso|.m1v|.m2v|.m2t|.m2ts|.m4v|.mkv|.mov|.mp2|.mp2v|.mp4|.mp4v|.mpe|.mpeg|.mpeg1|.mpeg2|.mpeg4|.mpg|.mpv2|.mts|.mtv|.mxf|.mxg|.nsv|.nuv|.ogg|.ogm|.ogv|.ogx|.ps|.rec|.rm|.rmvb|.rpl|.thp|.tod|.tp|.ts|.tts|.txd|.vob|.vro|.webm|.wm|.wmv|.wtv|.xesc)$');
```
同前

Commit:
fix(regex): fix vidRegex's escape letter
